2017-01-02 22:41:15,338 [INFO] from application in ForkJoinPool-1-worker-1 - Creating Pool for datasource 'default'
2017-01-02 22:41:20,598 [INFO] from play.api.db.HikariCPConnectionPool in ForkJoinPool-1-worker-1 - datasource [default] bound to JNDI as DefaultDS
2017-01-02 22:41:20,622 [INFO] from play.api.db.DefaultDBApi in ForkJoinPool-1-worker-1 - Database [default] connected at jdbc:mysql://localhost/jpa_sample?autoReconnect=true&useSSL=false
2017-01-02 22:41:24,697 [INFO] from play.api.Play in ForkJoinPool-1-worker-1 - Application started (Dev)
2017-01-02 22:41:25,460 [ERROR] from application in application-akka.actor.default-dispatcher-4 - 

! @72hmo4oo5 - Internal server error, for (GET) [/find-employees] ->
 
play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[RuntimeException: No EntityManager found in the context. Try to annotate your action method with @play.db.jpa.Transactional]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:293)
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:220)
	at play.api.GlobalSettings$class.onError(GlobalSettings.scala:160)
	at play.api.DefaultGlobal$.onError(GlobalSettings.scala:188)
	at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:100)
	at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:100)
	at play.core.server.netty.PlayRequestHandler$$anonfun$2$$anonfun$apply$1.applyOrElse(PlayRequestHandler.scala:99)
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344)
	at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:70)
	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)
	at scala.concurrent.impl.Promise$DefaultPromise.link(Promise.scala:304)
	at scala.concurrent.impl.Promise$DefaultPromise.linkRootOf(Promise.scala:289)
	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:253)
	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply$mcV$sp(BatchingExecutor.scala:91)
	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
	at akka.dispatch.BatchingExecutor$BlockableBatch$$anonfun$run$1.apply(BatchingExecutor.scala:91)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:72)
	at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:90)
	at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:39)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:415)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.RuntimeException: No EntityManager found in the context. Try to annotate your action method with @play.db.jpa.Transactional
	at play.db.jpa.JPAEntityManagerContext.em(JPAEntityManagerContext.java:30)
	at play.db.jpa.DefaultJPAApi.em(DefaultJPAApi.java:84)
	at com.knoldus.dao.impl.GenericDaoImpl.getEntityManager(GenericDaoImpl.java:25)
	at com.knoldus.dao.impl.GenericDaoImpl.findAll(GenericDaoImpl.java:41)
	at com.knoldus.service.impl.EmployeeServiceImpl.findAllEmployees(EmployeeServiceImpl.java:33)
	at com.knoldus.controllers.EmployeeController.findAllEmployees(EmployeeController.java:62)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$2$$anonfun$apply$2.apply(Routes.scala:137)
	at router.Routes$$anonfun$routes$1$$anonfun$applyOrElse$2$$anonfun$apply$2.apply(Routes.scala:137)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:157)
	at play.core.routing.HandlerInvokerFactory$$anon$4.resultCall(HandlerInvoker.scala:156)
	at play.core.routing.HandlerInvokerFactory$JavaActionInvokerFactory$$anon$14$$anon$3$$anon$1.invocation(HandlerInvoker.scala:136)
	at play.core.j.JavaAction$$anon$1.call(JavaAction.scala:73)
	at play.http.HttpRequestHandler$1.call(HttpRequestHandler.java:54)
	at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
	at play.core.j.JavaAction$$anonfun$7.apply(JavaAction.scala:108)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at play.core.j.HttpExecutionContext$$anon$2.run(HttpExecutionContext.scala:56)
	at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:70)
	at play.core.j.HttpExecutionContext.execute(HttpExecutionContext.scala:48)
	at scala.concurrent.impl.Future$.apply(Future.scala:31)
	at scala.concurrent.Future$.apply(Future.scala:492)
	at play.core.j.JavaAction.apply(JavaAction.scala:108)
	at play.api.mvc.Action$$anonfun$apply$2$$anonfun$apply$5$$anonfun$apply$6.apply(Action.scala:112)
	at play.api.mvc.Action$$anonfun$apply$2$$anonfun$apply$5$$anonfun$apply$6.apply(Action.scala:112)
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
	at play.api.mvc.Action$$anonfun$apply$2$$anonfun$apply$5.apply(Action.scala:111)
	at play.api.mvc.Action$$anonfun$apply$2$$anonfun$apply$5.apply(Action.scala:110)
	at scala.Option.map(Option.scala:146)
	at play.api.mvc.Action$$anonfun$apply$2.apply(Action.scala:110)
	at play.api.mvc.Action$$anonfun$apply$2.apply(Action.scala:103)
	at scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:251)
	... 14 common frames omitted
2017-01-02 22:46:21,624 [INFO] from application in play-fork-run-akka.actor.default-dispatcher-3 - Shutting down connection pool.
2017-05-24 11:08:53,651 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:08:54,150 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:00,120 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:09:00,124 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:00,716 [INFO] from application in pool-1-thread-1 - Shutting down connection pool.
2017-05-24 11:09:00,843 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:09:00,844 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:01,050 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:09:01,052 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:01,174 [INFO] from application in pool-1-thread-1 - Shutting down connection pool.
2017-05-24 11:09:01,311 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:09:01,314 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:01,486 [INFO] from application in pool-1-thread-1 - Creating Pool for datasource 'default'
2017-05-24 11:09:01,487 [INFO] from play.api.db.HikariCPConnectionPool in pool-1-thread-1 - datasource [default] bound to JNDI as TestDS
2017-05-24 11:09:01,648 [INFO] from application in pool-1-thread-1 - Shutting down connection pool.
